libkbfs/folder_branch_ops.go

```diff
@@ -3264,6 +3281,10 @@ func (fbo *folderBranchOps) getAndApplyMDUpdates(ctx context.Context,
 		return err
 	}
 
+	if len(rmds) > 0 {
+		fbo.setLatestRemoteMergedRevision(rmds[len(rmds)-1].Revision)
```

Comment:
You can move this into `applyMDUpdatesLocked()`, so that the proper lock is already taken.